### PR TITLE
Zhiyao/observation

### PR DESCRIPTION
**Pull request by teezhiyao**

### CODE REVIEW

File: Simulator/player.py [hunk 1]
```diff
@@ -326,34 +327,41 @@ def generate_champion_vectors(self):
...
-        for index, item in enumerate(curr_champ.items):
-            item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
-            # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now
-            # first spot of each of the item is left empty for now
-            item_arr[index * 2] = float(item_index) / self.MAX_ITEMS_IN_SET
-        champion_info_array[6:] = item_arr
+        for ind, item in enumerate(curr_champ.items):
+            if ind <= 2:
+                item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now
```
Comment (silverlight6) [inline Simulator/player.py:358]: Look at the item_stats.py --> item_builds. You can use that dictionary of lists to get the components that you are looking for.

File: Simulator/player.py [hunk 2]
```diff
@@ -844,7 +852,9 @@ def return_item_from_bench(self, x):
             if self.bench[x].items:
                 # thiefs_glove_loc_always needs to be cleared even if there's not enough room on bench
                 if self.bench[x].items[0] == 'thiefs_gloves':
-                    self.thiefs_glove_loc.remove([x, -1])
+                    if x in self.thiefs_glove_loc:
```
Comment (silverlight6) [inline Simulator/player.py:855]: I'll let this be for now but someone else is working on this bug and will fix it on their end.

**Review by silverlight6 — APPROVED** [review]

File: Simulator/player.py (hunk 2 above)
Comment (teezhiyao) [reply Simulator/player.py:855]: Okay, i'll just revert the changes here then.

File: Simulator/player.py [hunk 3]
```diff
@@ -354,13 +354,18 @@ def generate_single_champion_vector(self, curr_champ, champion_info_array):
...
             if ind <= 2:
-                item_index = list(item_list.keys()).index(item) + 1  # Avoiding index 0 as index 0 is reserved for no items
-                # Hoping to have the observation in terms of components instead but not too sure if the components can be retrieved. Leaving 1 spot empty for now
-                # first spot of each of the item is left empty for now
-                item_arr[ind * 2] = float(item_index) / self.MAX_ITEMS_IN_SET
+                if item in item_builds.keys():
+                    component1, component2 = item_builds[item]
+                    component1_index = list(item_list.keys()).index(component1) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                    component2_index = list(item_list.keys()).index(component2) + 1  # Avoiding index 0 as index 0 is reserved for no items
+                    item_arr[ind] = float(component1_index) / self.MAX_ITEMS_IN_SET
```
Comment (silverlight6) [inline Simulator/player.py:361]: If you're doing it this way, MAX_ITEMS_IN_SET is going to be like 9. This will also break for items like reforger.

**Pull request merged**